pkg/schedule/schedulers/hot_region.go

```diff
@@ -1518,54 +1612,35 @@ func (bs *balanceSolver) createSplitOperator(regions []*core.RegionInfo, isTooHo
 	operators := make([]*operator.Operator, 0)
 
 	createFunc := func(region *core.RegionInfo) {
-		stats, ok := hotBuckets[region.GetID()]
-		if !ok {
-			hotSchedulerRegionBucketsNotHotCounter.Inc()
-			return
-		}
-		// skip if only one hot buckets exists on this region.
-		if len(stats) <= 1 && isTooHot {
-			hotSchedulerRegionBucketsSingleHotSpotCounter.Inc()
-			return
-		}
-		startKey, endKey := region.GetStartKey(), region.GetEndKey()
-		splitKey := make([][]byte, 0)
-		for _, stat := range stats {
-			if keyutil.Between(startKey, endKey, stat.StartKey) && keyutil.Between(startKey, endKey, stat.EndKey) {
-				if len(splitKey) == 0 {
-					splitKey = append(splitKey, stat.StartKey, stat.EndKey)
-					continue
-				}
-				// If the last split key is equal to the current start key, we can merge them.
-				// E.g. [a, b), [b, c) -> [a, c) split keys is [a,c]
-				// Otherwise, we should append the current start key and end key.
-				// E.g. [a, b), [c, d) -> [a, b), [c, d) split keys is [a,b,c,d]
-				if bytes.Equal(stat.StartKey, splitKey[len(splitKey)-1]) {
-					// If the region is too hot, we should split all the buckets to balance the load.
-					// otherwise, we should split the buckets that are too hot.
-					if isTooHot {
-						splitKey = append(splitKey, stat.EndKey)
-					} else {
-						splitKey[len(splitKey)-1] = stat.EndKey
-					}
-				} else {
-					splitKey = append(splitKey, stat.StartKey, stat.EndKey)
+		switch strategy {
+		case bySize:
+			if op := bs.splitBucketBySize(region); op != nil {
+				operators = append(operators, op)
+			}
+		case byLoad:
+			stats, ok := hotBuckets[region.GetID()]
+			// If only one bucket is hot, we can't split it into two regions.
+			if !ok {
+				hotSchedulerRegionBucketsNotHotCounter.Inc()
+				return
+			}
```

Comment:
It seems that `bySize` doesn't need `hotBuckets`, maybe it would be better to move it under the `byLoad` branch.
BTW, do you think it would be better to move all related `byLoad`s code into the `splitBucketsByLoad` function?

Reply:
I don't think so, the split strategy for them is different. Devide into two branch is more simple.

Reply:
Maybe it's better to use `sync.once` for `hotBuckets` in `byLoads` case